ert/data/record/_transformation.py

```diff
@@ -22,71 +24,80 @@ def _prepare_location(base_path: Path, location: Path) -> None:
 class RecordTransformation(ABC):
     @abstractmethod
     async def transform_input(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
+        self, record: Record, mime: str, runpath: Path, location: Path
     ) -> None:
         raise NotImplementedError("not implemented")
 
     @abstractmethod
-    async def transform_output(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
-    ) -> None:
+    async def transform_output(self, mime: str, location: Path) -> Record:
         raise NotImplementedError("not implemented")
 
 
 class FileRecordTransformation(RecordTransformation):
     async def transform_input(
-        self,
-        transmitter: RecordTransmitter,
-        mime: str,
-        runpath: Path,
-        location: Path,
+        self, record: Record, mime: str, runpath: Path, location: Path
     ) -> None:
         _prepare_location(runpath, location)
-        await transmitter.dump(runpath / location, mime)
+        await _save_record_to_file(record, runpath / location, mime)
 
-    async def transform_output(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
-    ) -> None:
-        await transmitter.transmit_file(runpath / location, mime)
+    async def transform_output(self, mime: str, location: Path) -> Record:
+        return await _load_record_from_file(location, mime)
 
 
 class TarRecordTransformation(RecordTransformation):
     async def transform_input(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
+        self, record: Record, mime: str, runpath: Path, location: Path
     ) -> None:
-        record = await transmitter.load()
         if isinstance(record, BlobRecord):
             with tarfile.open(fileobj=io.BytesIO(record.data), mode="r") as tar:
                 _prepare_location(runpath, location)
                 tar.extractall(runpath / location)
         else:
             raise TypeError("Record needs to be a BlobRecord type!")
 
-    async def transform_output(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
-    ) -> None:
-        blob_record = BlobRecord(data=await make_tar(runpath / location))
-        await transmitter.transmit_record(blob_record)
+    async def transform_output(self, mime: str, location: Path) -> Record:
+        return BlobRecord(data=await make_tar(location))
 
 
 class ExecutableRecordTransformation(RecordTransformation):
     async def transform_input(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
+        self, record: Record, mime: str, runpath: Path, location: Path
     ) -> None:
-        # pre-make bin folder if neccessary
+        # pre-make bin folder if necessary
         base_path = Path(runpath / _BIN_FOLDER)
```

Comment:
If `location` is an absolute path, this folder will not be used. Should we error if `location` is not absolute? 

(this question might belong in a new issue)

Reply:
Actually it should probably error if location is anything but a filename. These are executables that we are dumping into a folder that is added to path, so if they are not added the that exact folder they will not be available later.

Reply:
That is already checked.